homeassistant/components/zha/entity.py

```diff
@@ -94,266 +83,80 @@ def device_info(self) -> DeviceInfo:
             via_device=(DOMAIN, zha_gateway.state.node_info.ieee),
         )
 
-    @callback
-    def async_state_changed(self) -> None:
-        """Entity state changed."""
-        self.async_write_ha_state()
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return entity specific state attributes."""
+        if self.entity_data.entity.extra_state_attribute_names is not None:
+            state_data = self.entity_data.entity.state
+            extra_state_attributes = {}
+            for name in self.entity_data.entity.extra_state_attribute_names:
+                if name in state_data and state_data[name] is not None:
+                    extra_state_attributes[name] = state_data[name]
+            return extra_state_attributes
+        return None
```

Comment:
We generally do not allow integrations to blindly accept / set the extra state attributes directly.

Should we map this?